test/cpp/end2end/xds/xds_end2end_test_lib.cc

```diff
@@ -175,6 +176,11 @@ void XdsEnd2endTest::ServerThread::Serve(grpc_core::Mutex* mu,
     builder.experimental().set_drain_grace_time(
         test_obj_->xds_drain_grace_time_ms_);
     builder.AddListeningPort(server_address, Credentials());
+    // Allow gRPC Core's HTTP server to accept PUT requests for testing
+    // purposes.
+    builder.AddChannelArgument(
+        GRPC_ARG_INTERNAL_ONLY_DO_NOT_USE_UNLESS_YOU_HAVE_PERMISSION_FROM_ALLOW_GRPC_TEAM_ALLOW_BROKEN_PUT_REQUESTS,
```

Comment:
Instead of hard-coding this here, how about adding a parameter to `CreateBackends()` that will enable this, so that we can use it only in the tests that need it?

Reply:
Done!